comp/dogstatsd/listeners/udp_test.go

```diff
@@ -27,32 +31,98 @@ import (
 type listenerDeps struct {
 	fx.In
 
-	Config config.Component
-	PidMap pidmap.Component
+	Config    config.Component
+	PidMap    pidmap.Component
+	Telemetry telemetry.Component
 }
 
 func fulfillDepsWithConfig(t testing.TB, overrides map[string]interface{}) listenerDeps {
 	return fxutil.Test[listenerDeps](t, fx.Options(
+		telemetryimpl.MockModule(),
 		config.MockModule(),
 		pidmapimpl.Module(),
 		fx.Replace(config.MockParams{Overrides: overrides}),
 	))
 }
 
-func newPacketPoolManagerUDP(cfg config.Component) *packets.PoolManager {
-	packetPoolUDP := packets.NewPool(cfg.GetInt("dogstatsd_buffer_size"))
+func newPacketPoolManagerUDP(cfg config.Component, packetsTelemetryStore *packets.TelemetryStore) *packets.PoolManager {
+	packetPoolUDP := packets.NewPool(cfg.GetInt("dogstatsd_buffer_size"), packetsTelemetryStore)
 	return packets.NewPoolManager(packetPoolUDP)
 }
 
 func TestNewUDPListener(t *testing.T) {
 	deps := fulfillDepsWithConfig(t, map[string]interface{}{"dogstatsd_port": "__random__"})
-	s, err := NewUDPListener(nil, newPacketPoolManagerUDP(deps.Config), deps.Config, nil)
+	telemetryStore := NewTelemetryStore(nil, deps.Telemetry)
+	packetsTelemetryStore := packets.NewTelemetryStore(nil, deps.Telemetry)
+	s, err := NewUDPListener(nil, newPacketPoolManagerUDP(deps.Config, packetsTelemetryStore), deps.Config, nil, telemetryStore, packetsTelemetryStore)
+
 	assert.NotNil(t, s)
 	assert.Nil(t, err)
 
 	s.Stop()
 }
 
+func TestUDPListenerTelemetry(t *testing.T) {
+	port, err := getAvailableUDPPort()
+	require.Nil(t, err)
+	cfg := map[string]interface{}{}
+	cfg["dogstatsd_port"] = port
+	cfg["dogstatsd_non_local_traffic"] = false
+
+	packetChannel := make(chan packets.Packets)
+	deps := fulfillDepsWithConfig(t, cfg)
+	telemetryStore := NewTelemetryStore(nil, deps.Telemetry)
+	packetsTelemetryStore := packets.NewTelemetryStore(nil, deps.Telemetry)
+	s, err := NewUDPListener(packetChannel, newPacketPoolManagerUDP(deps.Config, packetsTelemetryStore), deps.Config, nil, telemetryStore, packetsTelemetryStore)
+	require.NotNil(t, s)
+
+	assert.Nil(t, err)
+
+	s.Listen()
+	defer s.Stop()
+
+	fmt.Println(fmt.Sprintf("127.0.0.1:%d", port))
+	conn, err := net.Dial("udp", fmt.Sprintf("127.0.0.1:%d", port))
+	defer conn.Close()
+	assert.Nil(t, err)
```

Comment:
This should come before `defer`, and use `testify/require` to stop the test instead.